File: src/pressure_plate.ts
import { Coord, Dir } from "./coord";

export class PressurePlate {
    private pos: Coord;

    constructor(pos: Coord) {
        this.pos = pos;
    }

    is_on() {
        return `block ${this.pos.raw()} minecraft:polished_blackstone_pressure_plate[powered=true]`;
    }

    is_off() {
        return `block ${this.pos.raw()} minecraft:polished_blackstone_pressure_plate[powered=false]`;
    }
}

File: mcdsl/src/main/kotlin/com/zacklukem/mcdsl/PressurePlate.kt
package com.zacklukem.mcdsl

class PressurePlate(private val pos: Coord) {
    fun isOn(): Condition {
        return con("block $pos minecraft:polished_blackstone_pressure_plate[powered=true]")
    }

    fun isOff(): Condition {
        return con("block $pos minecraft:polished_blackstone_pressure_plate[powered=false]")
    }
}